internal/fs/buffer/inmemory_write_buffer_test.go

```diff
@@ -41,24 +46,240 @@ func (t *MemoryBufferTest) SetUp(ti *TestInfo) {
 
 func (t *MemoryBufferTest) TearDown() {}
 
+// //////////////////////////////////////////////////////////////////////
+// Helpers
+// //////////////////////////////////////////////////////////////////////
+
+func generateRandomData(size int64) []byte {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	data := make([]byte, size)
+	r.Read(data)
+	return data
+}
+
 // //////////////////////////////////////////////////////////////////////
 // Tests
 // //////////////////////////////////////////////////////////////////////
 
 func (t *MemoryBufferTest) TestCreateEmptyInMemoryBuffer() {
 	t.mb = CreateInMemoryWriteBuffer()
 
-	AssertEq(nil, t.mb.buffer)
+	AssertEq(nil, t.mb.currentBuffer)
+	AssertEq(nil, t.mb.flushedBuffer)
 	AssertEq(0, ChunkSize)
 }
 
 func (t *MemoryBufferTest) TestInitializeInMemoryBuffer() {
-	bufferSizeMB := 10
+	sizeInMB := 1
 	t.mb = CreateInMemoryWriteBuffer()
 
-	t.mb.InitializeBuffer(bufferSizeMB)
+	t.mb.InitializeBuffer(sizeInMB)
+
+	AssertEq(sizeInMB*MiB, ChunkSize)
+	AssertEq(sizeInMB*MiB, cap(t.mb.currentBuffer))
+	AssertEq(0, len(t.mb.currentBuffer))
+	AssertEq(0, t.mb.currentBufferStartOffset)
+	AssertEq(sizeInMB*MiB, t.mb.currentBufferEndOffset)
+
+	AssertEq(sizeInMB*MiB, cap(t.mb.flushedBuffer))
+	AssertEq(0, len(t.mb.flushedBuffer))
+	AssertEq(0, t.mb.flushedBufferStartOffset)
+	AssertEq(0, t.mb.flushedBufferEndOffset)
+}
+
+func (t *MemoryBufferTest) TestSingleWriteToInMemoryBuffer() {
+	// Allocate a buffer
+	t.TestInitializeInMemoryBuffer()
+	data := []byte("Taco")
+
+	// Write to buffer
+	err := t.mb.WriteAt(data, 0)
+
+	AssertEq(nil, err)
+	AssertEq(t.mb.fileSize, 4)
```

Comment:
Let's define 4 a const above in the setup section of the test or use `len(data)` in place of 4 here for easier maintenance.